sailor/pai/utils.py

```diff
@@ -0,0 +1,63 @@
+"""Module for various utility functions, in particular those related to fetching data from remote oauth endpoints."""
+
+import logging
+import warnings
+
+from ..utils.oauth_wrapper import get_oauth_client
+from ..utils.config import SailorConfig
+from ..utils.utils import DataNotFoundWarning
+from ..assetcentral.utils import _compose_queries
+
+LOG = logging.getLogger(__name__)
+LOG.addHandler(logging.NullHandler())
+
+
+def _fetch_data(endpoint_url, unbreakable_filters=(), breakable_filters=()):
+    """Retrieve data from the pai service."""
+    filters = _compose_queries(unbreakable_filters, breakable_filters)
+    oauth_client = get_oauth_client('pai')
+
+    if not filters:
+        filters = ['']
+
+    result = []
+    for filter_string in filters:
+        params = {'$filter': filter_string} if filter_string else {}
+        params['$format'] = 'json'
+
+        endpoint_data = oauth_client.request('GET', endpoint_url, params=params)
+
+        if isinstance(endpoint_data, list):
+            result.extend(endpoint_data)
+        else:
+            result.append(endpoint_data)
+
+    if len(result) == 0:
+        warnings.warn(DataNotFoundWarning(), stacklevel=2)
+    return result
+
+
+def _pai_application_url():
+    """Return the Pai application URL from the SailorConfig."""
+    return SailorConfig.get('pai', 'application_url')
+
+
+class PaiEntity:
+    """Common base class for Pai entities."""
+
+    def __init__(self, ac_json: dict):
+        """Create a new entity."""
+        self.raw = ac_json
+
+    def __repr__(self) -> str:
+        """Return a very short string representation."""
+        name = getattr(self, 'name', getattr(self, 'short_description', None))
```

Comment:
the Alert response only has "Description". we must map the 'short_description' to 'Description' in the mapping of the Alert class.

however, we should evaluate if we actually need a "global" PaiEntity for our PAI package. The PAI API does not have to offer a lot besides Alerts and some Timeseries access. 

Maybe we should abstract the AssetcentralEntity even further and reuse as much as possible in the Alerts class. Maybe that's for a different PR though. Thoughts @atrejut ?

Reply:
yes, I absolutely see the current AssetCentralEntity moving out into some `common` module, and would expect that for now we can have a new AssetCentralEntity and PAIEntity as subclasses of the common entity, but don't expect those subclasses to need any changes from the base class for now.

Reply:
would short_description  be more like type_description?

Reply:
PAI provide only alert type description, which might not be helpful. description is left out.